pkg/controller/installation/validation.go

```diff
@@ -92,24 +92,24 @@ func validateCustomResource(instance *operatorv1.Installation) error {
 				return fmt.Errorf("ipPool.CIDR(%s) is invalid: %s", v4pool.CIDR, err)
 			}
 
-			valid := false
-			for _, t := range operatorv1.EncapsulationTypes {
-				if v4pool.Encapsulation == t {
-					valid = true
+			// Verify the specified encapsulation type is valid.
+			switch v4pool.Encapsulation {
+			case operatorv1.EncapsulationIPIP, operatorv1.EncapsulationIPIPCrossSubnet:
+				// IPIP currently requires BGP to be running in order to program routes.
+				if instance.Spec.CalicoNetwork.BGP == nil || *instance.Spec.CalicoNetwork.BGP == operatorv1.BGPDisabled {
+					return fmt.Errorf("IPIP encapsulation requires that BGP is enabled")
 				}
-			}
-			if !valid {
+			case operatorv1.EncapsulationVXLAN, operatorv1.EncapsulationVXLANCrossSubnet:
+			case operatorv1.EncapsulationNone:
```

Comment:
Since it is currently invalid to have non-Calico CNI and CalicoNetwork, the case of EncapsulationNone also needs to have BGP enabled, I think.

Reply:
This might change soon with adding IPAM HostLocal as an option.